domain/model/service/service.go

```diff
@@ -26,25 +31,71 @@ type State interface {
 // Service defines a service for interacting with the underlying state based
 // information of a model.
 type Service struct {
-	st State
+	st         State
+	toolFinder ToolFinder
+}
+
+// ToolFinder represents a helper for establishing if tools for a specific Juju
+// are available.
+type ToolFinder interface {
```

Comment:
ToolsFinder

Also, `Tools` is a deprecated term. We should be using `Agent` or `AgentBinary`.

Reply:
Opted for agent binary as agent finder could mean a few things in this context.